lib/psi.rb

```diff
@@ -654,12 +654,117 @@ def generateKiva(os_model, walls, floors, edges)
         next unless s.outsideBoundaryCondition.downcase == "foundation"
         found = true
 
-        # Retrieve surface (standard) construction
+        # Retrieve surface (standard) construction before creating an exposed
+        # perimeter object.
+        construction = s.construction.get
         s.setAdjacentFoundation(floors[id][:foundation])
-        s.setConstruction(s.construction.get)
-        if floors[id][:exposed] > 0.01
-          s.createSurfacePropertyExposedFoundationPerimeter(arg, floors[id][:exposed])
-        end
+        s.setConstruction(construction)
+        exp = floors[id][:exposed]
+        exp = 0.01 if exp < 0.01
```

Comment:
This seems to work. The subsequent (lengthy) comments below (embedded in the code) give an idea of what occurs when setting the exposed perimeter to 0.0m or not setting the exposed perimeter at all if 0.0m. I think this points to a bug in OS-to-EPlus translation ...

Reply:
See [fix](https://github.com/rd2/tbd/commit/271ba7c25abef6c382f8b42784efde01c727fa5f#r45730611). Thanks for the suggestion @macumber.